### ISSUE TITLE
“hunt & scavenge” for open data in “smaller” languages

### ISSUE THREAD

**Issue by oepen** [OpenEuroLLM/training-data-catalogue#43]
[D3.1: Initial Catalogue and Analytics Reports for Existing Training Datasets](https://github.com/OpenEuroLLM/training-data-catalogue/blob/main/2507.deliverable.pdf)

(0) Target Languages: In our first deliverable (attached below), we summarized what is readily available from common web-derived datasets.  Please see Tables 2.1 and 2.2 for the current set of target languages for the project, and in particular those in the lower tiers of available data, in the bottom of Table 2.2.  We suggest we focus our target data acquisition efforts in T3.2 on at least the nine languages for which there are less than one billion of tokens readily available.

(1) Data Volume: Our WP4 colleagues tell us that they would like to include (at least) 100 billion tokens of training data for each language in the mix.  It can work to “repeat” training instances 2–4 times, so 25 billion arguably could be the lower mark we are aiming for.  This could include synthetic data (from T3.2), but the bottom line for T3.2 is that we are looking for “bulk”, at least tens or hundreds of millions of tokens to make it seem worthwhile to go after individual sources.

(2) Data Quality: For LLM pre-training, we are looking at full documents of “running text”, not word lists, shuffled sentences, translation memories, or the like.  What constitutes “good text quality” is of course a matter of debate, but one lesson appears to be that broad diversity actually is desirable.  Just carefully edited news, for example, would probably be valuable, but by itself it would be too lacking in variation.  For some LLM tasks, older and “outdated” data can be less valuable.

(3) Transparency and Replicability: For the OpenEuroLLM models to be open, we take it that others need to be able to inspect the training data that was used and, in principle, replicate our work.  That means we are looking for data that can be made “accessible” to others, including but not limited to European researchers.  As is the case for web-derived data, we are prepared to navigate lack of clarity regarding licensing, but broad accessibility (at no cost) is a basic principle.

**Comment by gramirez-prompsit** [comment]
Proposal on how to record new sources of data + quick view of web datasets: https://docs.google.com/spreadsheets/d/1ERMeyCK1gKepeToE2TkwSuv_xYyQbggwaCYIIp3k-Y0/edit?usp=sharing 

Proposal for metadata (all mandatory except size if not known): 

- Code: 3-letter language code
- Macro: macrolanguage 3-letter language code
- Script: language script
- Language: written name of language
- Variant: written name of language variant
- Domain: choose one of available, add new if necessary. 
- Dataset name or description: short and concise
- Date/date range: date of publication/online consultation, range of covered period (year-year)
- Size (Tokens): if available
- Size (Docs): if available
- Size (GB): if available
- License: licence name
- Source: publisher
- Formats: pdf, docx, pptx, etc. 
- Download link: link for downloading the documents/dataset
- Comments: general free field for comments
- Worth gathering? (1-maybe not, 2-maybe yes, 3-definitely): based on size, difficulty to download, clarity of licensing.
- In OpenEuroLLM Catalogue?: yes / no

**Comment by gramirez-prompsit** [comment]
Note: 
- Prompsit started downloading documents from selected sources. We will upload documents (PDFs, docx) to HPCs for processing. 
- Are we interested in transcrips or transcription of videos licensed with permissive licenses: https://vimeo.com/1012459924

**Comment by gramirez-prompsit** [comment]
**Data hunting guide and repo:** https://github.com/OpenEuroLLM/non_web_data. 

Summary: 

- 7 languages completed (Galician, Norwegian Nynorsk, Catalan, Georgian, Icelandic, Basque, Albanian)
- 12 TB downloaded, 1,235,210 assets, 47971 hours of audio stored, > 12,295,348,153 word tokens 

**Text extraction from hunted data (documents and audio) repo:** https://github.com/OpenEuroLLM/extract_pdf_audio. 

Summary: 
- currently testing pdf extraction pipelines
- currently testing audio extraction pipelines
